verification/curator-service/api/test/model/automation.test.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    AutomationDocument,
+    automationParsingValidator,
+    automationSchema,
+} from '../../src/model/automation';
+
+import { Error } from 'mongoose';
+import fullModel from './data/automation.full.json';
+import regexParsingModel from './data/regex-parsing.full.json';
+import mongoose from 'mongoose';
+
+const Automation = mongoose.model<AutomationDocument>(
+    'Automation',
+    automationSchema,
+);
+
+// Used to test automationParsingValidator.
+const wrapperSchema = new mongoose.Schema({
+    automation: {
+        type: automationSchema,
+        validate: automationParsingValidator,
+    },
+});
+type WrapperDocument = mongoose.Document & { automation: AutomationDocument };
+const Wrapper = mongoose.model<WrapperDocument>('Wrapper', wrapperSchema);
+
+describe('validate', () => {
+    it('an automation without a schedule is invalid', async () => {
+        const missingSchedule = { ...fullModel };
+        delete missingSchedule.schedule;
+
+        return new Automation(missingSchedule).validate((e) => {
+            expect(e.name).toBe(Error.ValidationError.name);
+        });
+    });
+
+    it('an automation without either a parser or regexParsing is invalid', async () => {
```

Comment:
should there be a similar one for the inverse? (remove regexParser but leave parser)?

Reply:
The loaded object contains only a parser -- so what we're testing is that neither exist. The full model contains a parser but no regexParsing. I can add a test for an object with just a regexParsing.